runtime/queries/go/context.scm

```diff
@@ -0,0 +1,17 @@
+; Credits to nvim-treesitter/nvim-treesitter-context
+
+(function_declaration
+	result: (_) @context.end
+) @context
+
+(method_declaration
+	result: (_) @context.end
```

Comment:
function_declaration and method_declaration 

`result:(_)` --> `body:(_)`

in my test
```
[
  (function_declaration) 
  (method_declaration)
  (if_statement)
  (for_statement)
  ...
]@context
```
has same result.

`if_statement`, `for_statement`, `const_declaration`, rendering the contents of the next line, this should be related to the 'non-ascii' mentioned earlier.

Also, in golang std library , like `src/strconv/itoa.go` , `if_statement`, `for_statement` render error.

Reply:
https://github.com/golang/go/blob/master/src/strconv/itoa.go
![WX20230324-122434](https://user-images.githubusercontent.com/716514/227424045-7b74c775-3da5-4398-9f86-ab593da3602a.png)

Reply:
> if_statement, for_statement, const_declaration, rendering the contents of the next line, this should be related to the 'non-ascii' mentioned earlier.

i don't think this is the case anymore, but i think go context queries might be a bit different.
I will investigate it.
Also if you have any ideas with how to improve the go queries, feel free to open a pr on my pr-branch :)

Reply:
I'm sorry that there are still similar errors in the previous test file. I didn't see anything in the log.

I has delete this branch and re pull this pr.
rev: 4fa00ec6bf48765a30a9ff6b680c1da7ceed42fd
test file: https://gist.github.com/erasin/a15c4f250beff8c9f7e4aaeae4a1128c

Reply:
yeah you're right.

Investigating now..

Reply:
thank for fix.

Performance problem, after using this pr, when the page scrolls, the cpu usage rate is 1-2 times higher.
db3e4e8cb1ead6e18d8a9c9328b29904c87257ec

Reply:
> Performance problem, after using this pr, when the page scrolls(use Zj or mouse), the cpu usage rate is 1-2 times higher.

With mouse the performance issues are kinda expected as the sticky context does a lot of calculations and is rather expensive.
I could think about some more caching ideas, but that gonna be... Difficult.

Reply:
I forgot add `--release`, after using release , it is much better. 

thanks for the work.